prow/git/v2/client_factory.go

```diff
@@ -287,6 +306,17 @@ func (c *clientFactory) ClientFromDir(org, repo, dir string) (RepoClient, error)
 // org and repo are used for determining where the repo is cloned, cloneURI
 // overrides org/repo for cloning.
 func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
+	// These RepoOpts are the default, and is here to preserve behavior of
+	// existing callsites that did not previously know about
+	// ClientForWithRepoOpts.
+	repoOpts := RepoOpts{
+		SparseCheckoutDirs:         nil,
+		ShareObjectsWithSourceRepo: false,
+	}
```

Comment:
This is actually just RepoOpts{}, and it could be inlined in the call below.

Reply:
I did this originally to make things more explicit (to show the reader "hey, we're not using these options"). I'm OK with inlining it. Will update.

Reply:
Done.